src/main/kotlin/network/rs485/logisticspipes/config/ServerConfigurationManager.kt

```diff
@@ -78,11 +78,21 @@ class ServerConfigurationManager {
         return internalRepresentation.playerConfigurations.keys
     }
 
-    fun getPlayerConfiguration(identifier: PlayerIdentifier): ClientConfiguration? {
-        return internalRepresentation.playerConfigurations[identifier] ?: ClientConfiguration()
+    fun getPlayerConfiguration(identifier: PlayerIdentifier): PlayerConfiguration {
+        return internalRepresentation.playerConfigurations[identifier] ?: PlayerConfiguration()
     }
 
     fun setClientConfiguration(identifier: PlayerIdentifier, configuration: ClientConfiguration) {
+        val newConfigurations = HashMap(internalRepresentation.playerConfigurations)
+        if(newConfigurations[identifier] == null) {
+            newConfigurations[identifier] = PlayerConfiguration()
+        }
+        newConfigurations[identifier]?.merge(configuration)
```

Comment:
should be a `!!` because it must exist at this point.

Reply:
Or `newConfigurations.computeIfAbsent(identifier, {PlayerConfiguration()}).merge(configuration)`